lib/cache/cache.go

```diff
@@ -517,18 +723,28 @@ func (c *Cache) notify(event Event) {
 //   we assume that this cache will eventually end up in a correct state
 //   potentially lagging behind the state of the database.
 //
-func (c *Cache) fetchAndWatch(ctx context.Context, retry utils.Retry) error {
+func (c *Cache) fetchAndWatch(ctx context.Context, retry utils.Retry, timer *time.Timer) error {
 	watcher, err := c.Events.NewWatcher(c.ctx, services.Watch{
 		QueueSize:       c.QueueSize,
 		Name:            c.Component,
 		Kinds:           c.watchKinds(),
 		MetricComponent: c.MetricComponent,
 	})
 	if err != nil {
-		c.notify(Event{Type: WatcherFailed})
+		c.notify(c.ctx, Event{Type: WatcherFailed})
 		return trace.Wrap(err)
 	}
 	defer watcher.Close()
+
+	// ensure that the timer is stopped and drained
```

Comment:
Why not just initialize this here?

Reply:
Initializing it before the `update()` loop lets us re-use the timer when the cache resets.  Not that timers are terribly expensive, but IIUC its best practice to re-use them in long-running loops.